src/MealDetail/MealDetail.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import {withRouter} from 'react-router';
+
+import './MealDetail.less';
+
+class MealDetail extends React.Component {
+	constructor() {
+		super();
+
+		this.state = {
+			name: "",
+			prepHours: 0,
+			prepMinutes: 0,
+			taste: 5,
+			difficulty: 0
+		}
+	}
+
+	handleSave = () => {
+		const prepTime = parseInt(this.state.prepMinutes) + (parseInt(this.state.prepHours) * 60);
+		const { name, taste, difficulty } = this.state;
+		const newMeal = {
+			name,
+			prepTime,
+			taste,
+			difficulty
+		};
+		this.props.saveHandler(newMeal);
+		this.props.history.push('/');
+	};
+
+	render() {
+		return (
+			<div id="main" className="container">
+				<div className="row">
+					<div className="col-xs-12 col-md-4">
+						<input
+							type="text"
+							className="form-control"
+							value={ this.state.name }
+							onChange={ (event) => this.setState({name: event.target.value}) }/>
```

Comment:
Similar comment to the above, binding this way generates a new function every time the component is rendered. It is probably better to extract this into:
```javascript
handleNameChange = (event) => { this.setState({ name: event.target.value }) }
```
Then in the `render()`, you'd bind to `onChange={this.handleNameChange}`